runtime/caches/redis.ts

```diff
@@ -47,13 +39,33 @@ export const caches: CacheStorage = {
   ): Promise<Response | undefined> => {
     throw new Error("Not Implemented");
   },
-  open: (cacheName: string): Promise<Cache> => {
-    if (!redis) {
-      throw new Error(
-        "Redis coult not be used due to the lack of credentials.",
-      );
-    }
-    const requestURLSHA1 = withCacheNamespace(cacheName);
+  open: async (_: string): Promise<Cache> => {
```

Comment:
an important note: 

Open receives `cacheName` as a string, this should be used to "shard" your cache, so ideally you need to use to compose your cache keys to avoid namespace keys clash.

also: open can be called multiple times. a common approach is to create a redis at module level var

let redis: ReturnType<typeof createClient> | null = null;

then here you can just use

redis ??= createClient({ url: REDIS_URL })...

in that way you're going to create a single instance even multiple opens were called
```suggestion
  open: async (_: string): Promise<Cache> => {
```